src/coreclr/jit/valuenum.cpp

```diff
@@ -8726,6 +8726,18 @@ static bool GetStaticFieldSeqAndAddress(ValueNumStore* vnStore, GenTree* tree, s
         }
     }
 
+    if (tree->IsCnsIntOrI())
+    {
+        FieldSeq* fldSeq = tree->AsIntCon()->gtFieldSeq;
+        if ((fldSeq != nullptr) && (fldSeq->GetKind() == FieldSeq::FieldKind::SimpleStaticKnownAddress))
```

Comment:
Can we `AddToFieldAddressToFieldSeqMap` for all `SimpleStaticKnownAddress` constants instead to avoid this special case?

I think we've mentioned that case in the VN->field seq map change, but (mistakenly) believe it to not be worth handling.

Reply:
@SingleAccretion thanks, did I address your feedback?